### DISCUSSION TITLE
SellingApiForbiddenException -

### DISCUSSION THREAD

**Discussion by Prasand14** [saleweaver/python-amazon-sp-api#477] [Q&A]
I could retrieve the same data from Postman without any access issue. Can someone help me where i'm going wrong.
Thanks in Advance 

Code Used
`
from sp_api.api import Sales
from sp_api.base import Granularity
from sp_api.api import Orders

credentials=dict(
        refresh_token = SP_REFRESH_TOKEN,
        lwa_app_id = SP_CLIENT_ID,
        lwa_client_secret = SP_CLIENT_SECRET,
        aws_secret_key = SP_SECRET_KEY,
        aws_access_key = SP_ACCESS_KEY,
        role_arn = role_arn
    )

interval = ('2022-03-22T00:00:00-07:00--2022-03-23T00:00:00-07:00')

response = Sales(credentials=credentials,marketplace=Marketplaces.US).get_order_metrics(interval, Granularity.TOTAL, granularityTimeZone='US/Central')

`
Error Statement
`SellingApiForbiddenException: [{'message': 'Access to requested resource is denied.', 'code': 'Unauthorized', 'details': ''}]`

**Comment by denisneuf** [comment]
I think you are having same problem:
Get Order Metrics Interval Issue

```
from sp_api.api import Sales
from sp_api.base import Granularity
res = Sales(credentials=credentials).get_order_metrics(
interval=("2021-02-20T00:00:00+08:00", "2022-02-20T00:00:00+08:00"),
granularity = Granularity.DAY)
```

Please read this:

https://github.com/saleweaver/python-amazon-sp-api/discussions/407

**Comment by Prasand14** [comment]
Hi Denis,
Thanks for taking your time helping. But i'm getting the same error.
![Screenshot 2022-03-31 134715](https://user-images.githubusercontent.com/15247021/161010003-1e93324d-5e55-47b0-a62f-3daa3b4123e3.jpg)

**Comment by denisneuf** [comment]
I am sorry as I just tested the code in my side and is working fine.
Seems your problem is related with your credentials so I could not help you too much on it. Though there is some discussions that some people make some mistake with lwa_app_id using the Amazon App ID instead of the Client identifier in the LWA credentials which looks alike as it start with "amzn1.[.....]".
Did you tried any other end point to discard that situation?
